private/rules/jvm_import.bzl

```diff
@@ -8,15 +8,14 @@
 # [1]: https://github.com/bazelbuild/bazel/issues/4549
 
 load("@rules_java//java:defs.bzl", "JavaInfo")
+load("//private/lib:bzlmod.bzl", "to_visible_name")
 load("//settings:stamp_manifest.bzl", "StampManifestProvider")
 
 def _jvm_import_impl(ctx):
     if len(ctx.files.jars) != 1:
         fail("Please only specify one jar to import in the jars attribute.")
 
-    # With `bzlmod` enabled, workspace names end up being `~` separated. For the
-    # user-visible workspace name, we need the final part of the name
-    visible_name = ctx.label.workspace_name.rpartition("~")[2]
+    visible_name = to_visible_name(ctx.label.workspace_name)
```

Comment:
@fmeum can this usage of `to_visible_name` be avoided? the label is added as _part_ of a string arg.

Reply:
(can't comment on the lines below)

```patch
-progress_message = "Stamping the manifest of %s" % ctx.label,
+progress_message = "Stamping the manifest of %{label}",
```

```patch
-args.add_all(["--manifest-entry", "Target-Label:{target_label}".format(target_label = label)])
+args.add("--manifest-entry", label, format = "Target-Label:%s")
```

Both should give you apparent labels with Bazel 7. The are also more memory efficient as the formatted string isn't retained.

Reply:
Done.